src/c++/library/common.h

```diff
@@ -334,6 +334,15 @@ class InferInput {
   /// \return Error object indicating success or failure.
   Error AppendFromString(const std::vector<std::string>& input);
 
+  /// Get access to the buffer holding raw input. Note the buffer is owned by
+  /// InferInput instance. Users can copy out the data if required to extend
+  /// the lifetime.
+  /// \param buf Returns the pointer to the start of the buffer.
+  /// \param byte_size Returns the size of buffer in bytes.
+  /// \return Error object indicating success or failure of the
+  /// request.
+  Error RawData(const uint8_t** buf, size_t* byte_size);
```

Comment:
Are you sure this one is needed? This isn't part of the other InferInput class tree (and isn't implementing a virtual function). I think this is triton specific and is never going to be used by perf_analyzer, but I may be wrong.

Reply:
Unlike other endpoints (e.g. `OpenAiInferInput`, `TorchServeInferInput` or `TFServeInferInput`) that contain raw buffer of input inside the class, `TritonInferInput` class does not contain the raw buffer. Instead, it has a pointer to the Triton client `InferInput` object (in common.h), which under the hood has the raw data. This triton client class has an accessor functions like `GetNext()` but it's private and using it shifts buffer index internally, which I would like to avoid. So I just added a separate getter called `RawData()` similar to how `InferResult` class does to access

Reply:
I understand that, but I don't think that this class is ever held inside of PA. I think it is only used in the client-backend to directly interface with Triton, and it doesn't need this new function. I may be wrong though.

Reply:
How else would `TritonInferInput` object get access to the raw data? The class itself does not hold the raw data, so all it can do is ask [`input_`](https://github.com/triton-inference-server/client/pull/549/files#diff-5382697538d808eef2f4411869634f594f6cb1d6492a036dc6df505cc8dc15edR293) variable to return one.